cli/args/mod.rs

```diff
@@ -1702,8 +1702,13 @@ fn warn_insecure_allow_run_flags(flags: &Flags) {
     return;
   };
 
-  // discourage using --allow-run without an allow list
-  if allow_run_list.is_empty() {
+  let is_no_args_deno = match flags.subcommand.clone() {
+    DenoSubcommand::Repl(repl_flags) => repl_flags.is_default_command,
+    _ => false,
+  };
+  // discourage using --allow-run without an allow list, but
+  // allow it when no args deno is used [deno] without subcommand
+  if allow_run_list.is_empty() && !is_no_args_deno {
```

Comment:
I think we should change this to instead set `--allow-all`: https://github.com/denoland/deno/blob/f360cae9dd1150c1ca02d51631ddaddb5bf871ab/cli/args/flags.rs#L1370-L1376

Reply:
There's actually a `flags.allow_all();` and I think internally that should just set `flags.permissions.allow_all = true` only.